vrs/marten-construction-script.py

```diff
@@ -1,168 +1,230 @@
-# ^^^ TEMPORARY PATH
+multiallelic_path = tmp_path + "multiallelic.ht"
+ht_multiallelic = ht_multiallelic.checkpoint(
+    multiallelic_path, _read_if_exists=True, overwrite=True
+)  # , overwrite=overwrite, _read_if_exists=read_if_exists)
 
 # Construct table of variants altered by min-rep
-
 # code in this section provided by Julia Goodrich (best boss this side of the Mississippi)
 vds = hl.vds.read_vds("gs://gnomad/v3.1/raw/gnomad_v3.1.vds")
 ht_minrep = vds.variant_data.rows()
+# ht_vds = hl.vds.read_vds(vds_path).variant_data.rows() # never told path, revert back to original
 ht_minrep = ht_minrep.annotate(original_alleles=ht_minrep.alleles)
 ht_minrep = hl.split_multi(ht_minrep)
-ht_minrep = ht_minrep.filter(ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index])
-
-ht_minrep_filt = ht_minrep._filter_partitions(sample(range(ht_minrep.n_partitions()), 5))
-ht_minrep_filt = ht_minrep_filt.select()
-ht_minrep_final = ht_minrep_filt.join(ht_whole, how = "inner")
+ht_minrep = ht_minrep.filter(
+    ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index]
+)
+ht_minrep = ht_minrep._filter_partitions(sample(range(ht_minrep.n_partitions()), 5))
+ht_minrep = ht_minrep.select()
+ht_minrep = ht_minrep.join(ht_whole, how="inner")
+# an inner join here joins on Locus and Allele
+# doing the same using a collect() and filter() step would be awkward, since it involves two separate fields
+# doing the following
 # ^ results in some few thousand variants, all being samples changed by min-rep
 
 # checkpoint for final minrep variants
-minrep_path_final = "gs://gnomad-tmp-4day/marten-temp-01-06/minrep-checkpoint.ht"
-ht_minrep_final = ht_minrep_final.checkpoint(minrep_path_final, overwrite = True)
-
-minrepcount = ht_minrep_final.count()
-print(f'There are {minrepcount} variants in the minrep set')
+minrep_path_final = tmp_path + "minrep-checkpoint.ht"
+ht_minrep = ht_minrep.checkpoint(
+    minrep_path_final, _read_if_exists=True, overwrite=True
+)
 
-# Final output and construction steps
+# Counts and further assembly
+minrepcount = ht_minrep.count()
+print(f"There are {minrepcount} variants in the minrep set")
 
 ht_prefinal = ht_union.union(ht_multiallelic)
-ht_final = ht_prefinal.union(ht_minrep_final)
-final_path = "gs://gnomad-tmp/marten-temp-01-06/final_output.ht"
-ht_final = ht_final.checkpoint(final_path,overwrite = True)
-# this also writes the table to where you want it, so you can set this as the final output!!!
-
-ht_final.show(5)
-ht_final.describe()
-ht_final.count()
-
-## spiked-in Extra ERRORS Coming Soon !!! 
+ht_final = ht_prefinal.union(ht_minrep)
+
+# Check for duplicates and de-duplication
+
+# Simplified de-duplication
+final_count = ht_final.count()
+ht_final_distinct = ht_final.distinct()
+duplicates_removed = final_count - ht_final_distinct.count()
+print(f"Number of duplicates removed: {duplicates_removed}")
+ht_final = ht_final_distinct
```

Comment:
can also add a logger for the final total variant count after removing duplicates